OpenEdXMobile/src/main/java/org/edx/mobile/view/custom/CropImageView.java

```diff
@@ -108,7 +108,9 @@ protected void onDraw(Canvas canvas) {
         mRectF.set((float) canvas.getWidth() / 2 - radius, (float) canvas.getHeight() / 2 - radius, (float) canvas.getWidth() / 2 + radius, (float) canvas.getHeight() / 2 + radius);
         circleSelectionPath.reset();
         circleSelectionPath.addOval(mRectF, Path.Direction.CW);
-        canvas.clipPath(circleSelectionPath, Region.Op.XOR);
+
+        canvas.clipPath(circleSelectionPath, Region.Op.DIFFERENCE);
```

Comment:
Instead of using this deprecated function where there is margin for error, we can use `clipOutPath` function which does the same thing.

_copied from source code:_
```java
    /**
     * Set the clip to the difference of the current clip and the specified path.
     *
     * @param path The path used in the difference operation
     * @return     true if the resulting clip is non-empty
     */
    public boolean clipOutPath(@NonNull Path path) {
        return clipPath(path, Region.Op.DIFFERENCE);
    }
```

Reply:
Agree

Reply:
p.s. you'll have to add an API Level 26 check for this function as it was introduced in it:
https://developer.android.com/reference/android/graphics/Canvas#clipOutPath(android.graphics.Path)

Reply:
Already done 😄